joblib/memory.py

```diff
@@ -853,52 +752,85 @@ class Memory(Logger):
 
-    def __init__(self, cachedir, mmap_mode=None, compress=False, verbose=1,
-                 bytes_limit=None):
+    def __init__(self, location=None, backend='local', cachedir=None,
+                 mmap_mode=None, compress=False, verbose=1, bytes_limit=None,
+                 store_options={}):
         """
             Parameters
             ----------
-            cachedir: string or None
+            location: str or None
+                The path of the base directory to use as a data store
+                or None. If None is given, no caching is done and
+                the Memory object is completely transparent. This option
+                replaces cachedir since version 0.11.
+
+            backend: str or 'local'
+                Type of store backend for reading/writing cache files.
+                Default is 'local'.
+
+            cachedir: str or None
+                cachedir is deprecated since version 0.11 and will be
+                removed in 0.13. Please consider using location option instead.
                 The path of the base directory to use as a data store
                 or None. If None is given, no caching is done and
                 the Memory object is completely transparent.
+
             mmap_mode: {None, 'r+', 'r', 'w+', 'c'}, optional
                 The memmapping mode used when loading from cache
                 numpy arrays. See numpy.load for the meaning of the
                 arguments.
+
             compress: boolean, or integer
                 Whether to zip the stored data on disk. If an integer is
                 given, it should be between 1 and 9, and sets the amount
                 of compression. Note that compressed arrays cannot be
                 read by memmapping.
+
             verbose: int, optional
                 Verbosity flag, controls the debug messages that are issued
                 as functions are evaluated.
+
             bytes_limit: int, optional
-                Limit in bytes of the size of the cache
+                Limit in bytes of the size of the cache.
+
         """
         # XXX: Bad explanation of the None value of cachedir
         Logger.__init__(self)
         self._verbose = verbose
         self.mmap_mode = mmap_mode
         self.timestamp = time.time()
-        self.compress = compress
         self.bytes_limit = bytes_limit
+        self.backend = backend
         if compress and mmap_mode is not None:
             warnings.warn('Compressed results cannot be memmapped',
                           stacklevel=2)
-        if cachedir is None:
-            self.cachedir = None
-        else:
-            self.cachedir = os.path.join(cachedir, 'joblib')
-            mkdirp(self.cachedir)
+        if cachedir is not None:
+            if location is None:
+                warnings.warn("cachedir option is deprecated since version "
+                              "0.10 and will be removed after version 0.12.\n"
```

Comment:
I am not sure what the versions should be but at the moment they are inconcistent in this PR. This one says 0.10 -> 0.12. Other places say 0.11 -> 0.13.

I am guessing it would make sense to release a 0.12 with all the changes that are going into master at the moment.